internal/sqlparser/parser_test.go

```diff
@@ -453,3 +453,69 @@ func isCIEnvironment() bool {
 	ok, _ := strconv.ParseBool(os.Getenv("CI"))
 	return ok
 }
+
+func TestErrorHandling(t *testing.T) {
+	t.Parallel()
+
+	type testData struct {
+		name     string
+		sql      string
+		errCheck func(error) bool
+	}
+
+	tests := []testData{
+		{
+			name: "MissingSemicolon",
+			sql:  syntaxMissingSemicolons,
+			errCheck: func(err error) bool {
+				return strings.Contains(err.Error(), "missing semicolon")
+			},
+		},
+		{
+			name: "InvalidStatementBegin",
+			sql:  syntaxInvalidStmtBegin,
+			errCheck: func(err error) bool {
+				expectedErr := newGooseError(ErrInvalidStatementBeginFormat, 0)
+				return err.Error() == expectedErr.Error()
+			},
+		},
+		{
+			name: "DuplicateGooseUp",
+			sql:  syntaxDuplicateGooseUp,
+			errCheck: func(err error) bool {
+				expectedErr := newGooseError(ErrDuplicateGooseUpFormat, gooseUp)
+				return err.Error() == expectedErr.Error()
+			},
+		},
+		{
+			name: "InvalidStatementEnd",
+			sql:  syntaxInvalidStmtEnd,
+			errCheck: func(err error) bool {
+				expectedErr := newGooseError(ErrInvalidStatementEndFormat)
+				return err.Error() == expectedErr.Error()
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, _, err := ParseSQLMigration(strings.NewReader(test.sql), DirectionUp, false)
+			if err == nil || !test.errCheck(err) {
```

Comment:
If we remove the GooseError, we can assert directly for the sentinel error with `errors.Is` ?